### PR TITLE
Feature/lookup table docs

### PR DESCRIPTION
**Pull request by halfhorst**
I find it difficult to word this and also difficult to draw a line between lookup and interpolation. Let me know if you want more or less of that here, or an example with just a scalar, etc.

Additionally, should we start adding glossary terms? Do we ever prefer linking to those over the actual concepts / tutorials?

### CODE REVIEW

File: src/vivarium/framework/lookup.py [hunk 1]
```diff
@@ -257,38 +260,44 @@ class LookupTableInterface:
     def __init__(self, manager):
         self._lookup_table_manager = manager
 
-    def build_table(self, data, key_columns=('sex',), parameter_columns=(['age', 'age_group_start', 'age_group_end'],
-                                                                         ['year', 'year_start', 'year_end']),
-                    value_columns=None) -> LookupTable:
+    def build_table(self, data,
+                    key_columns: Union[List[str], Tuple[str]]=('sex',),
+                    parameter_columns: Union[List[str], Tuple[str]]=(['age', 'age_group_start', 'age_group_end'],
```
Comment (halfhorst) [inline src/vivarium/framework/lookup.py]: we have a choice here. This typing is not currently correct. Iterable and Sequence can't distinguish a string from an iter/seq of strings (https://stackoverflow.com/questions/44912374/python-type-annotation-for-sequences-of-strings-but-not-for-strings). To be exact, the typing info would be insane. We could make a type var or leave it inexact in two ways, with iterable or precluding the list of list situation with bin edges (which is always)

Comment (kfwilson) [reply src/vivarium/framework/lookup.py]: hmm yeah the parameter columns typing should be list/tuple of lists/tuples of strings. I would tend to not go for iterable and just use lists/tuples and make a typevar so it's not insanely long in the function def. I'm a little confused about what you mean by precluding the list of list situation though?

File: docs/source/concepts/lookup.rst [hunk 2]
```diff
@@ -1,16 +1,150 @@
...
+      2     30.0
+      3     27.0
+      4     25.0
+      Name: BMI, dtype: float64
+
+.. note::
+    Constructing a lookup table currently requires your data meet specific
+    conditions. These are a consequence of the method the lookup table uses to
+    arrive at the correct data. Specifically, your parameter columns must
+    represent bins and they must overlap.
```
[the remark this answers is not in the document]
Comment (halfhorst) [reply docs/source/concepts/lookup.rst]: ok that's good, I should be very specific with what overlap means, I'll address that

File: docs/source/concepts/lookup.rst [hunk 3]
```diff
@@ -1,16 +1,150 @@
...
+population in a simulation.
+
+Construction Parameters
+~~~~~~~~~~~~~~~~~~~~~~~
+
+A lookup table is defined for a set of categorical variables, continuous
+variables, and the values that depend on those variables. The lookup table
+calls these variables keys, parameters, and values, respectively.
+
+key
```
[the remark this answers is not in the document]
Comment (halfhorst) [reply docs/source/concepts/lookup.rst]: I think I can't get away without a scalar table example, I'll add one

**Pull request merged**